rfc/sigil.md

```diff
@@ -0,0 +1,822 @@
+
+```json
+// alice.manager links the whole user:bob object
+{
+  "the": "application/json",
+  "of": "user:alice",
+  "is": {
+    "name": "Alice Smith",
+    "manager": {
+      "/": {
+        "link@1": {
+          "accept": "application/json",
+          "source": "user:bob"
+        }
+      }
+    }
+  }
+}
+
+// alice.manager.name = "Charlie" always mutates user:bob
+{
+  "the": "application/json",
+  "of": "user:bob",
+  "is": {
+    "name": "Charlie",  // Target fact's property modified
+    "department": "Engineering",
+    "role": "Manager"
+  }
+}
+```
+
+### Use Cases
+
+- **`overwrite: "this"`**: Use when you want a mutable reference that can be replaced with direct values
+- **`overwrite: "redirect"`**: Use when you want transparent redirection, like permanent redirects or symbolic links
+
+### Binary Content Sigils
+
+Binary content sigils are references that need to be interpreted as `Blob` instances. They can use either immutable references for content that never changes, or mutable references (link sigils) for binary data that can be updated.
+
+#### Blob Sigil (`blob@1`)
+
+The blob sigil provides references to binary data that should be interpreted as a `Blob` instance. It can reference either immutable content via immutable references or mutable binary facts via link sigils.
+
+#### Fields
+
+- `type` (optional): Media type of the binary data. If omitted and content is referenced by immutable reference, it will be inferred from content as either `application/json` or `application/octet-stream`. If content is referenced by link sigil, content type will be the type of resolved data.
+- `content` (required): Either a link sigil referencing a fact, an immutable reference to immutable content, or inline binary data using bytes representation
+
+#### TypeScript Definition
+
+```typescript
+type BlobSigil = {
+  "blob@1": {
+    type?: MIME // defaults to application/octet-stream
+    content: LinkSigil | Reference | Bytes
+  }
+}
+
+type Reference = {
+  "/": string // Merkle reference to fact `is` field content
+}
+
+type Bytes = {
+  "/": {
+    bytes: string // base64-encoded binary data
+  }
+}
+```
+
+#### Resolution Behavior
+
+When `content` is inline bytes, the binary data is embedded directly and the content type defaults to `application/octet-stream` if no `type` field is specified.
+
+#### Relationship to Binary Facts
+
+Blob sigils work directly with the memory protocol's binary fact support by referencing facts that contain binary data. This approach enables efficient storage and reuse of binary content across multiple references while maintaining proper media type information.
+
+#### Example with Link Reference
+
+```json
+{
+  "the": "application/json",
+  "of": "user:alice",
+  "is": {
+    "name": "Alice Smith",
+    "avatar": {
+      "/": {
+        "blob@1": {
+          "type": "image/png",
+          "content": {
+            "/": {
+              "link@1": {
+                "accept": "image/png",
```

Comment:
ah the accept & type parts are a bit redundant here.

i know you said you went back and forth, but why a LinkSigil instead of a URI? I.e. an alternative could be `content: URI | Reference | Bytes` and maybe add an optional `space` attribute that specifies where to find the `URI` or the `Reference` (default for `URI` is same space and for `Reference` I suppose any other mechanism, though maybe we have a default place things associated with this space would go, where we can look first)